tee-worker/litentry/core/vc-task/sender/src/lib.rs

```diff
@@ -37,13 +36,16 @@ use std::{
 
 #[derive(Debug)]
 pub struct VCRequest {
-	pub sender: oneshot::Sender<Result<Vec<u8>, String>>,
+	pub sender: MpscSender<Result<Vec<u8>, String>>,
 	pub request: AesRequest,
 }
 
-#[derive(Encode, Decode, Clone)]
-pub struct VCResponse {
-	pub vc_payload: Vec<u8>,
+#[derive(Debug, Encode, Decode, Clone)]
+pub struct RequestVcResultOrError {
```

Comment:
I actually prefer using enum to represent "result or error"

Reply:
Maybe we should create an issue and solve it in another PR - in case you don't have any capacity for this one